src/lib/error/bad-data-error.ts

```diff
@@ -1,23 +1,90 @@
-class BadDataError extends Error {
-    constructor(message: string) {
-        super();
-        Error.captureStackTrace(this, this.constructor);
+import { ErrorObject } from 'ajv';
+import { getPropFromString } from '../util/get-prop-from-string';
+import { ApiErrorSchema, UnleashError } from './api-error';
 
-        this.name = this.constructor.name;
-        this.message = message;
+type ValidationErrorDescription = {
+    description: string;
+    message: string;
+    path?: string;
+};
+class BadDataError extends UnleashError {
+    details: ValidationErrorDescription[];
+
+    constructor(
+        message: string,
+        errors?: [ValidationErrorDescription, ...ValidationErrorDescription[]],
+    ) {
+        const topLevelMessage =
+            'Request validation failed: your request body contains invalid data' +
+            (errors
+                ? '. Refer to the `details` list to see what happened.'
+                : `: ${message}`);
+        super(topLevelMessage);
+
+        this.details = errors ?? [{ message: message, description: message }];
     }
 
-    toJSON(): object {
+    toJSON(): ApiErrorSchema {
         return {
-            isJoi: true,
-            name: this.constructor.name,
-            details: [
-                {
-                    message: this.message,
-                },
-            ],
+            ...super.toJSON(),
+            details: this.details,
         };
     }
 }
 
 export default BadDataError;
+
+export const fromOpenApiValidationError =
```

Comment:
do we have any tests describing what this function does? The logic is quite involved and I'd appreciate some examples showing how a given input translates to a given output.